src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -303,18 +303,15 @@ const GraphComponent *GafferScene::EditScopeAlgo::transformEditReadOnlyReason( c
 		return MetadataAlgo::readOnlyReason( rows );
 	}
 
-	// As any of our columns could be locked, pick the first one.
-	for( const auto child : { g_translate, g_rotate, g_scale, g_pivot } )
+	for( const auto &plug : Plug::RecursiveRange( *row->cellsPlug() ) )
 	{
-		const Spreadsheet::CellPlug *cell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( child );
-		if( !cell ) { continue; }
-
-		if( const auto reason = MetadataAlgo::readOnlyReason( cell->valuePlug<V3fPlug>() ) )
+		if( MetadataAlgo::getReadOnly( plug.get() ) )
 		{
-			return reason;
+			return MetadataAlgo::readOnlyReason( plug.get() );
```

Comment:
Why not `return plug.get()`? We've checked the ancestors via the recursion, so why check them all again via `readOnlyReason()`? Is it because we haven't checked `rows` itself? If so, maybe it makes sense to do a single check on that before hitting the recursion?

Reply:
`readOnlyReason` should always return the outward most `GraphComponent` that is causing read-only-ness. We haven't checked any of the the ancestors of `row->cellsPlug()` yet. Checking that first would save the recursion if you're worried about the cost of it.

Reply:
See [update](https://github.com/GafferHQ/gaffer/pull/3840/commits/c669a77363b8fd831db6ddd4a18866d533852c0c)

Reply:
Thanks - that's better. It was partially the excess cost that bothered me, but also the logic seemed strange - "if _this_ plug has been made read only, then we should search for _other_ reasons it might be read only", followed by "we didn't actually find any plugs that were themselves read only, which means we didn't do any searches, so now we should do a search".